File: src/main/java/com/shopmanager/service/SettingsService.java
package com.shopmanager.service;

import com.shopmanager.dto.settings.SettingsRequest;
import com.shopmanager.dto.settings.SettingsResponse;
import org.springframework.stereotype.Service;

@Service
public interface SettingsService {
    SettingsResponse getSettings();
    SettingsResponse updateSettings(SettingsRequest request);
    Object testWhatsAppConnection();
    Object triggerBackup();
}

File: api/settingsService.js
import axiosClient from './axiosClient';

const settingsService = {

    getSettings: async () => {
        const response = await axiosClient.get('/api/settings');
        return response.data;
    },

    updateSettings: async (settingsData) => {
        const response = await axiosClient.put('/api/settings', settingsData);
        return response.data;
    },
};

export default settingsService;